src/util-flow-rate.c

```diff
@@ -172,7 +171,10 @@ static inline void FlowRateStoreUpdateCurrentRing(
         DEBUG_VALIDATE_BUG_ON(frs->dir[direction].sum < prev_byte_count);
         /* Sum should get rid of previous count on the same index */
         frs->dir[direction].sum += pkt_len - prev_byte_count;
-        frs->dir[direction].start_ts = p_ts;
+        if (idx != frs->dir[direction].last_idx + 1) {
```

Comment:
Is the comment on line 167:  `/* Index is revisited after a full round of the buffer */`

Still valid now that the else covers more cases?